pkg/secrets/vault.go

```diff
@@ -291,30 +291,48 @@ func (decrypter *VaultDecrypter) newAPIClient() (*api.Client, error) {
 
 func (decrypter *VaultDecrypter) fetchSecret(client VaultClient) (string, error) {
 	path := decrypter.engine + "/" + decrypter.path
-	log.Debugf("attempting to read secret at KV v1 path: %s", path)
-	secretMapping, err := client.Read(path)
-	if err != nil {
-		if strings.Contains(err.Error(), "invalid character '<' looking for beginning of value") {
+	log.Infof("attempting to read secret at KV v1 path: %s", path)
+	secretMapping, v1err := client.Read(path)
+	if v1err != nil {
+		if strings.Contains(v1err.Error(), "invalid character '<' looking for beginning of value") {
```

Comment:
Minor:
That message makes it look like the underlying error is an [encoding/json SyntaxError](https://golang.org/pkg/encoding/json/#SyntaxError) from the VaultClient. I'm not sure if it is possible to get the underlying error type here, but this would be helpful compared to string matching.

```
if e, ok := v1err.(*json.SyntaxError); ok {
		return "", fmt.Errorf("error fetching secret from vault - check connection to the server: %s", decrypter.vaultConfig.Url)
}
```

Reply:
Hmm yeah, I'll look into that. It should be easy to recreate!

Reply:
^ this totally worked and agree that it's cleaner. It would be nice to not even have that kind of misleading error, but it's coming from inside the vault API, so I guess this is good enough. It's not a super common error anyways